superset-frontend/src/components/DvtSelect/dvt-select.module.tsx

```diff
@@ -0,0 +1,102 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { styled } from '@superset-ui/core';
+
+interface StyledSelectProps {
+  isOpen: boolean;
+}
+
+interface StyledSelectOptionProps {
+  selectedValue: string;
+  value: string;
+}
+
+const StyledSelect = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const StyledSelectSelect = styled.div<StyledSelectProps>`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding: 12px;
+  width: 202px;
+  height: 48px;
+  border-radius: 12px;
+  background-color: ${({ isOpen, theme }) =>
+    isOpen
+      ? theme.colors.dvt.primary.light2
+      : theme.colors.dvt.grayscale.light2};
+  border: none;
+  appearance: none;
+  margin-bottom: 3px;
+  cursor: pointer;
+  color: ${({ theme }) => theme.colors.dvt.primary.light1};
+  transition: background-color 0.3s ease-in-out;
+`;
+
+const StyledSelectLabel = styled.label`
+  padding-left: 13px;
+  font-weight: 600;
+`;
+
+const StyledSelectOption = styled.div<StyledSelectOptionProps>`
+  padding: 13px;
+  cursor: pointer;
+  ${({ theme, value, selectedValue }) =>
+    value === selectedValue
+      ? `
+        color: ${theme.colors.grayscale.light5};
+        background: ${theme.colors.dvt.primary.light1};
+        &:first-of-type {
+          border-radius: 12px 12px 0px 0px;
+        }
+        &:last-of-type {
+          border-radius: 0px 0px 12px 12px;
+        }
+      `
+      : `
+        color: ${theme.colors.dvt.primary.light1};
+      }
+    `}
+`;
+const StyledSelectOptions = styled.div<StyledSelectProps>`
+  width: 202px;
+  border-radius: 12px;
+  background: ${({ theme }) => theme.colors.dvt.primary.light2};
+  max-height: ${({ isOpen }) => (isOpen ? '274px' : '0')};
+  overflow-y: auto;
+  transition: max-height 0.3s ease-in-out;
+`;
```

Comment:
StyledSelectOptions > position absolute olmalı. eğer absolute olmazsa sayfadaki 5 tane select sıralanması olursa open close olursa height sorun çıkarabilir...

position: absolute; olacak top, left taşınma halledeceksin.
max-height: ${({ isOpen }) => (isOpen ? '274px' : '0')}; olmamalı. Bu sorun var option 2 item varsa kaydırmasız ama açıp kapatınca hep kaydırma gözüküyor sorun olur.  sadece max-height: 274px bırakmalısın ve

index.tsx  {isOpen && <StyledSelectOptions .... }  böyle daha kolay olur.

transition: max-height 0.3s ease-in-out; aslında animation ve keyframes transform scaleY(0) ve scaleY(1) koyup yeterli

transform-origin: top; verirsen bu animasyon desteği verir